suite-common/wallet-core/src/token-definitions/tokenDefinitionsThunk.ts

```diff
@@ -0,0 +1,56 @@
+import { createThunk } from '@suite-common/redux-utils';
+import { Network } from '@suite-common/wallet-config';
+
+import { selectSpecificTokenDefinition } from './tokenDefinitionsReducer';
+import { tokenDefinitionsActions } from './tokenDefinitionsActions';
+
+export const getTokenDefinition = createThunk(
+    `${tokenDefinitionsActions.actionsPrefix}/getTokenDefinition`,
+    async (params: { network: Network; contractAddress: string }, { getState, dispatch }) => {
+        const { network, contractAddress } = params;
+        const tokenDefinition = selectSpecificTokenDefinition(
+            network.symbol,
+            contractAddress,
+        )(getState());
+
+        let isTokenKnown = tokenDefinition?.isTokenKnown;
+
+        // if token was never fetched and is not loading right now
+        if (typeof isTokenKnown === 'undefined' && !tokenDefinition?.isLoading) {
+            try {
+                dispatch(
+                    tokenDefinitionsActions.fetchInit({
+                        networkSymbol: network.symbol,
+                        contractAddress,
+                    }),
+                );
+
+                const fetchedTokenDefinition = await fetch(
+                    `https://data.trezor.io/firmware/eth-definitions/chain-id/${
+                        network.chainId
+                    }/token-${contractAddress.substring(2).toLowerCase()}.dat`,
+                    { method: 'HEAD' },
+                );
+
+                isTokenKnown = fetchedTokenDefinition.ok;
+
+                dispatch(
```

Comment:
Don't create extra actions for success/fail thunks. RTK has built-in support for these https://redux-toolkit.js.org/api/createAsyncThunk#examples and https://redux-toolkit.js.org/api/createAsyncThunk#handling-thunk-errors

Reply:
1392726d11396c5c98f19050f898470c9fb3f939